src/VisualStudio/Core/Impl/LanguageServiceBrokerShim/LanguageServiceBrokerShim.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Threading;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageServiceBrokerShim
+{
+    // Microsoft.VisualStudio.LanguageServer.Client.Implementation does not ship on nuget but Microsoft.VisualStudio.LanguageServices does so we cannot depend
+    // on it directly - we instead need this shim to act as a redirect.
```

Comment:
It appears there was some misunderstanding here: you can add the reference to the regular VS package -- you don't need this interface shim. We absolutely do upload Microsoft.VisualStudio.LanguageServices to NuGet, but we have a custom nuspec for it that overrrides the list of dependencies that nuget.org will know about:

https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/Core/Def/Microsoft.VisualStudio.LanguageServices.nuspec

If you want to defer this to a later PR, that's fine, but if you do track this in the mop-up list somewhere.

Reply:
Interface shim has been removed 👍